### PR TITLE
fix: recover when there is unreleased lock file

### PR DESCRIPTION
**Pull request by sdowell**
If a previous git invocation crashes, it is possible that an orphaned lock file (e.g. shallow.lock) is left on the filesystem. This previously caused git-sync to crash loop because the lock file is never deleted.

This change adds a check in sanityCheckRepo for the existence of a git lock file. If the git lock file exists at this stage, then initRepo will re-initialize the repository.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -1227,6 +1227,19 @@ func (git *repoSync) removeStaleWorktrees() (int, error) {
 	return count, nil
 }
 
+func hasGitLockFile(gitRoot absPath) (bool, error) {
```
Comment (thockin) [inline main.go:1230]: How about also returning the name of the first detected lockfile which , so the caller can log it?

**Comment by thockin** [comment]
Just one super small point, then LGTM.  Thanks!

File: main.go [hunk 2]
```diff
@@ -1258,6 +1272,16 @@ func (git *repoSync) sanityCheckRepo(ctx context.Context) bool {
 		return false
 	}
 
+	// Check if the repository contains an unreleased lock file. This can happen if
+	// a previous git invocation crashed.
+	if lockFile, ok, err := hasGitLockFile(git.root); err != nil {
```
[the remark this answers is not in the document]
Comment (sdowell) [reply main.go]: No worries, refactored

**Comment by thockin** [comment]
Thanks!

/lgtm
/approve

**Pull request merged**